### PR TITLE
Feat(eos_cli_config_gen): Add schema for ip_community_lists

### PR DESCRIPTION
**Pull request by ghost**
### Contributor Checklist

- [x] Create schema fragment matching data model described in README.md and README_v4.0.md
  - README.md is most complete with all keys. README_v4.0 includes partial data models after conversion to lists.
  - Schema fragment path is `roles/eos_cli_config_gen/schemas/schema_fragments/<data_model_key>.schema.yml`.
  - Copy line 1-5 from another schema (comments and outer type:dict).
  - Refer to [schema documentation](https://avd.sh/en/devel/docs/input-variable-validation-BETA.html) for syntax and/or use YAML Lint plugin from Redhat in VSCode.
  - Use `convert_types` on value that could be mixed type or misinterpreted like integers and numeric strings.
- [x] If the data model has been converted from wildcard dicts:
  - Add `convert_types: ['dict']` to the schema.
  - Remove `convert_dicts` from the `templates/eos/<>.j2` and `templates/documentation/<>.j2` templates.
- [x] Run `molecule converge -s build_schemas_and_docs` to update schema and documentation.
- [x] Test by running `molecule converge -s eos_cli_config_gen` and verify no errors or changes to generated configs/docs.

### Reviewer Checklist

- Reviewer 1 (Claus):
  - [x] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [ ] Verify that `convert_dicts` has been removed from templates as applicable.
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

- Reviewer 2:
  - [ ] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [ ] Verify that `convert_dicts` has been removed from templates as applicable.
  - [ ] Verify no changes to configs/docs on any molecule scenario
  - [ ] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        entries:
+          type: list
+          required: true
+          items:
+            type: dict
+            keys:
+              action:
+                type: str
+                required: true
+                description: |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml]: If these are the only two valid options (please check) then use `valid_values` to describe and enforce this.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                  "< permit or deny >"
+                  Example: "permit"
+              communities:
+                type: str
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml]: From the description this looks like a list of strings. Please change the schema accordingly.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                required: false
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml]: This is default and can be left out of the schema.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml (hunk 1 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml]: added

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/Input Variables.md [hunk 4]
```diff
@@ -36,3 +36,49 @@ access_lists:
...
+| [<samp>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;regexp</samp>](## "ip_community_lists.[].entries.[].regexp") | String |  |  |  | "< regular expression >"<br>If defined, regex community list will be configured |
+
+### YAML
+
+```yaml
+ip_community_lists:
+  - name: <str>
+    entries:
+      - action: <str>
+        communities:
```
Comment (Shivani-chourasiya) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/Input Variables.md:113]: @ClausHolbechArista ,is it expected like this? the type is not visible here

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/Input Variables.md:113]: it is not handled correctly no. The issue is that you have not created the schema for the list items. Please update that, and then you will see this correctly.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml]: but then you should remove the description

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            type: dict
+            keys:
+              action:
+                type: str
+                required: true
+                valid_values: ["permit", "deny"]
+                description: |
+                  "< permit or deny >"
+                  Example: "permit"
+              communities:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml:50]: please fix this schema. You are missing items.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+                type: list
+                description: |
+                  [ "< a_community as string >", "< another_community as string >", ... ]
+                  This is optional, if defined - standard community list will be configured
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml]:
```suggestion
                description: "If defined, a standard community-list will be configured"
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+              regexp:
+                type: str
+                description: |
+                  "< regular expression >"
+                  If defined, regex community list will be configured
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml]:
```suggestion

                display_name: Regular Expression
                description: "If defined, a regex community-list will be configured"
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_community_lists.schema.yml]: please remove the description

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by KonikaChaurasiya-GSLab — APPROVED** [review]
LGTM!

**Pull request merged**